src/mpsc/tests/mpsc_blocking.rs

```diff
@@ -452,3 +452,36 @@ fn tx_close_drains_queue() {
         producer.join().unwrap();
     });
 }
+
+#[test]
+fn test_full() {
```

Comment:
In order to properly reproduce issue #83, do we also need to add at least one "leaked" `RecvRef`, so that the slot-skipping behavior is exercised (similarly to the test you added here https://github.com/hawkw/thingbuf/blob/c11720e08aec1dc8978941547df8fd3fd8188066/src/lib.rs#L755-L778)? Or, will this test detect a deadlock if run against the current `main` branch?